lib/datadog/di/probe.rb

```diff
@@ -31,11 +31,17 @@ module DI
     #
     # @api private
     class Probe
+      KNOWN_TYPES = %i[log].freeze
+
       def initialize(id:, type:,
         file: nil, line_no: nil, type_name: nil, method_name: nil,
         template: nil, capture_snapshot: false, max_capture_depth: nil, rate_limit: nil)
         # Perform some sanity checks here to detect unexpected attribute
         # combinations, in order to not do them in subsequent code.
+        unless KNOWN_TYPES.include?(type)
+          raise ArgumentError, "Unknown probe type: #{type}"
+        end
+
         if line_no && method_name
           raise ArgumentError, "Probe contains both line number and method name: #{id}"
         end
```

Comment:
Out of curiosity, what happens if this error gets triggered? E.g. does the component that is applying probe configuration deal correctly with these exceptions?

(And here "correctly" is doing a lot of work -- I guess "correctly" would be to apply all the probes it can, ignore the ones it can't, and maybe not fill up the logs too much? Also, not try to apply more than once probes that were rejected?)

Reply:
Exception boundaries are listed in the RFC, the one applicable to this exception is in the remote config processor when it is processing probe specifications to install.

For not trying repeatedly, this is an interesting question. I added it to my list of questions to research.